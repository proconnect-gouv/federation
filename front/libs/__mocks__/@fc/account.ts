import React from 'react';

export const AccountProvider = jest.fn();

export const AccountContext = React.createContext(expect.any(Object));
