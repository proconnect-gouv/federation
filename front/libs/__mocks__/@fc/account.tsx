import React from 'react';

export const AccountProvider = jest.fn(({ children }) => (
  <div>
    <div>AccountProvider</div>
    <div>{children}</div>
  </div>
));

export const AccountContext = React.createContext(expect.any(Object));
