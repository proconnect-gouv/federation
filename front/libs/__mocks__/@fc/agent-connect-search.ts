import * as React from 'react';

export const AgentConnectSearchContext = React.createContext({
  hasSearched: false,
  payload: {
    csrfToken: 'csrf-token-mock',
    identityProviders: [],
    isLoaded: false,
    ministries: [],
    serviceProviderName: undefined,
  },
  searchResults: [],
  setSearchTerm: () => {},
});

export const useSearchResults = jest.fn(() => ({
  searchResults: [
    {
      identityProviders: [
        {
          active: true,
          display: true,
          name: 'mock-idp-name-1.1',
          uid: 'mock-idp-uid-1.1',
        },
        {
          active: true,
          display: true,
          name: 'mock-idp-name-1.2',
          uid: 'mock-idp-uid-1.2',
        },
      ],
      ministry: {
        id: 'mock-ministry-id-1',
        identityProviders: ['mock-idp-uid-1'],
        name: 'mock-ministry-name-1',
        sort: 1,
      },
    },
    {
      identityProviders: [
        {
          active: true,
          display: true,
          name: 'mock-idp-name-2.1',
          uid: 'mock-idp-uid-2.1',
        },
        {
          active: true,
          display: true,
          name: 'mock-idp-name-2.2',
          uid: 'mock-idp-uid-2.2',
        },
      ],
      ministry: {
        id: 'mock-ministry-id-2',
        identityProviders: ['mock-idp-uid-2'],
        name: 'mock-ministry-name-2',
        sort: 0,
      },
    },
  ],
  showNoResults: false,
  showResults: true,
}));

export const useOnSearch = jest.fn(() => jest.fn());
