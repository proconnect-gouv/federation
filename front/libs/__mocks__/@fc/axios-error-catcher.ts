import React from 'react';

export const AxiosErrorCatcherProvider = jest.fn();

export const AxiosErrorCatcherContext = React.createContext(expect.any(Object));
