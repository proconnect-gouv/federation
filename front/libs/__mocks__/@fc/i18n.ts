export const I18nService = {
  initialize: jest.fn(),
};

export const t = jest.fn((key) => key);
