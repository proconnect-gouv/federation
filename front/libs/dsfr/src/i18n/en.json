{
  "DSFR.stepper.location": "Step {current} of {total}",
  "DSFR.stepper.nextStep": "Next step"
}
