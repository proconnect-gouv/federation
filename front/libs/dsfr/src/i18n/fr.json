{
  "DSFR.stepper.location": "Étape {current} sur {total}",
  "DSFR.stepper.nextStep": "Étape suivante",
  "DSFR.breadcrumbs.show": "Voir le fil d’Ariane",
  "DSFR.breadcrumbs.location": "vous êtes ici :",
  "DSFR.button.loginConnect.whatIs": "Qu’est ce que {connectType} ?",
  "DSFR.button.loginConnect.authenticateWith": "S’identifier avec",
  "DSFR.alert.close": "Masquer le message",
  "DSFR.notice.close": "Masquer le message"
}
