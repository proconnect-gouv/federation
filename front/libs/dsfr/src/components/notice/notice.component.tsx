import React from 'react';

import type { NavigationLinkInterface } from '@fc/common';
import { t } from '@fc/i18n';

interface NoticeComponentProps {
  title: string;
  description?: string;
  onClose?: () => void;
  link?: NavigationLinkInterface;
}

export const NoticeComponent = React.memo(
  ({ description, link, onClose, title }: NoticeComponentProps) => {
    const closeLabel = (onClose && t('DSFR.notice.close')) || undefined;
    const linkTitle = link?.title ? `${link.title} - ${t('FC.Common.newWindow')}` : undefined;

    return (
      <div className="fr-notice fr-notice--info">
        <div className="fr-container">
          <div className="fr-notice__body">
            <p>
              <span className="fr-notice__title">{title}</span>
              {description && <span className="fr-notice__desc">{description}</span>}
              {link && (
                <a
                  className="fr-notice__link"
                  href={link.href}
                  rel="noopener external noreferrer"
                  target="_blank"
                  title={linkTitle}>
                  {link.label}
                </a>
              )}
            </p>
            {!!onClose && (
              <button className="fr-btn--close fr-btn" title={closeLabel} onClick={onClose}>
                {closeLabel}
              </button>
            )}
          </div>
        </div>
      </div>
    );
  },
);

NoticeComponent.displayName = 'NoticeComponent';
