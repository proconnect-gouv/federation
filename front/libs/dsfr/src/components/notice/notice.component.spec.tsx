import { fireEvent, render } from '@testing-library/react';

import { t } from '@fc/i18n';

import { NoticeComponent } from './notice.component';

describe('NoticeComponent', () => {
  it('should match snapshot', () => {
    // When
    const { container, getByText } = render(<NoticeComponent title="any-title-mock" />);
    const titleElt = getByText('any-title-mock');

    // Then
    expect(container).toMatchSnapshot();
    expect(titleElt).toBeInTheDocument();
  });

  it('should match snapshot with optionnals props', () => {
    // Given
    jest
      .mocked(t)
      .mockReturnValueOnce('DSFR.notice.close-mock')
      .mockReturnValueOnce('FC.Common.newWindow-mock');

    // When
    const { container, getByText, getByTitle } = render(
      <NoticeComponent
        description="any-description-mock"
        link={{
          href: 'any-href-mock',
          label: 'any-link-label-mock',
          title: 'any-link-title-mock',
        }}
        title="any-title-mock"
        onClose={jest.fn()}
      />,
    );
    const titleElt = getByText('any-title-mock');
    const linkElt = getByText('any-link-label-mock');
    const descriptionElt = getByText('any-description-mock');
    const buttonElt = getByTitle('DSFR.notice.close-mock');

    // Then
    expect(container).toMatchSnapshot();
    expect(t).toHaveBeenCalledTimes(2);
    expect(t).toHaveBeenNthCalledWith(1, 'DSFR.notice.close');
    expect(t).toHaveBeenNthCalledWith(2, 'FC.Common.newWindow');
    expect(titleElt).toBeInTheDocument();
    expect(descriptionElt).toBeInTheDocument();
    expect(linkElt).toBeInTheDocument();
    expect(linkElt).toHaveAttribute('href', 'any-href-mock');
    expect(linkElt).toHaveAttribute('title', 'any-link-title-mock - FC.Common.newWindow-mock');
    expect(buttonElt).toBeInTheDocument();
  });

  it('should call onClose with user click the button', () => {
    // Given
    const onCloseMock = jest.fn();
    jest.mocked(t).mockReturnValueOnce('DSFR.notice.close-mock');

    // When
    const { getByTitle } = render(<NoticeComponent title="any-title-mock" onClose={onCloseMock} />);
    const buttonElt = getByTitle('DSFR.notice.close-mock');
    fireEvent.click(buttonElt);

    // Then
    expect(onCloseMock).toHaveBeenCalledOnce();
    expect(t).toHaveBeenCalledOnce();
    expect(t).toHaveBeenCalledWith('DSFR.notice.close');
  });
});
