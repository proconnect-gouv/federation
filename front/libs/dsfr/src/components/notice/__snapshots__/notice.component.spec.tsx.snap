// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoticeComponent should match snapshot 1`] = `
<div>
  <div
    class="fr-notice fr-notice--info"
  >
    <div
      class="fr-container"
    >
      <div
        class="fr-notice__body"
      >
        <p>
          <span
            class="fr-notice__title"
          >
            any-title-mock
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`NoticeComponent should match snapshot with optionnals props 1`] = `
<div>
  <div
    class="fr-notice fr-notice--info"
  >
    <div
      class="fr-container"
    >
      <div
        class="fr-notice__body"
      >
        <p>
          <span
            class="fr-notice__title"
          >
            any-title-mock
          </span>
          <span
            class="fr-notice__desc"
          >
            any-description-mock
          </span>
          <a
            class="fr-notice__link"
            href="any-href-mock"
            rel="noopener external noreferrer"
            target="_blank"
            title="any-link-title-mock - FC.Common.newWindow-mock"
          >
            any-link-label-mock
          </a>
        </p>
        <button
          class="fr-btn--close fr-btn"
          title="DSFR.notice.close-mock"
        >
          DSFR.notice.close-mock
        </button>
      </div>
    </div>
  </div>
</div>
`;
